ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.ts

```diff
@@ -1552,6 +1764,97 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    // For Grand Octet:
+    // After bosses and dragons start spawning, there's no clear log line we can trigger off of to find bosses' position
+    // until it's effectively too late.  The best way to do this seems to be to fire the trigger
+    // with a delay when Bahamut uses Grand Octet before all 3 bosses jump.
+    {
+      id: 'UCU Grand Octet Run & Rotate',
+      type: 'Ability',
+      // Grab mob position data after dragons/bosses are positioned
+      netRegex: { id: '26E7', source: 'Bahamut Prime', capture: false },
+      delaySeconds: 4.8,
+      promise: async (data) => {
+        if (!data.trioSourceIds.nael || !data.trioSourceIds.bahamut)
+          return;
+        data.combatantData = [];
+        data.combatantData = (await callOverlayHandler({
+          call: 'getCombatants',
+          ids: [data.trioSourceIds.nael, data.trioSourceIds.bahamut],
+        })).combatants;
+      },
+      alertText: (data, _matches, output) => {
+        let naelDirIdx;
+        let bahaDirIdx;
+
+        for (const mob of data.combatantData) {
+          const mobDirIdx = Math.round(4 - 4 * Math.atan2(mob.PosX, mob.PosY) / Math.PI) % 8;
```

Comment:
Is this `positionTo8DirOutput`?

Reply:
`positionTo8DirOutput` returns the direction name from `dirNumToName`, but we actually need the array index (or, more precisely, the direction number), because we use the direction number in `naelDirIdx` and `bahaDirIdx` to calculate the safe spot (based on opposite-Bahamut, possible adjustment for Nael).  
If we wanted to use `positionTo8DirOutput` here instead of directly calling `Math.atan2`, we could grab the direction name, but then we'd have to do an `indexOf` lookup to get back to the direction number, which feels clunky.

Reply:
As an aside, given that we seem to use relatively common variants of positionTo8Dir now in various fights, perhaps these (and, optionally, their 4Dir and 16Dir variants?) would be good candidates to add to a function library, so we're not repeating this each time?  If you agree, I'd be happy to submit a PR, but not sure where would be the right place for them to live.

Reply:
Maybe it'd be better for `positionTo8DirOutput` to return a number, and you could look up from there and use it in all places? Personally I think it's fine to re-do the number=>string lookup, but the math part would be nice to keep in one spot.

Agreed that it'd be good to put these in common locations.  I think a good common location could just be `util.ts`.  We could make some new functions for "common raidboss functions" or "raidboss helpers" too (since watch combatants might be a better fit in there than util).  One problem is that a couple of fights have conflicting ideas of what 0 means (there's like ~2-3 that want to have 0=NW and most are 0=N) and so those would need to be refactored/renamed as well.

Reply:
Sounds good.  Let's pause this PR for now, and I'll open up a new one to add some functionality to util.  If & once you're good with it and land it, I can revise this PR to use the new functions and then do a follow-on PR for the other fights.